201-vmss-slb-ilb/prereqs/prereqs.azuredeploy.json

```diff
@@ -0,0 +1,83 @@
+{
+
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+
+  "contentVersion": "1.0.0.0",
+
+  "parameters": {
+
+  },
+
+  "variables": {
+
+    "vnetName": "vnet-01",
+
+    "vnetLocation": "West US",
+
+    "vnetPrefix": "10.0.0.0/24",
+
+    "vnetFESubnetName": "subnet01",
+
+    "vnetFESubnetPrefix": "10.0.0.0/25",
+    "publicIPAddressName": "vmsspip"    
+
+  },
+
+  "resources": [
+    {
+      "name": "[variables('vnetName')]",
+      "type": "Microsoft.Network/virtualNetworks",
+      "location": "[variables('vnetLocation')]",
+      "apiVersion": "2017-10-01",
+
+      "dependsOn": [],
+      "tags": {
+        "displayName": "vnet"
+      },
+
+      "properties": {
+        "addressSpace": {
+          "addressPrefixes": [
+            "[variables('vnetPrefix')]"
+          ]
+        },
+
+        "subnets": [
+          {
+            "name": "[variables('vnetFESubnetName')]",
+            "properties": {
+              "addressPrefix": "[variables('vnetFESubnetPrefix')]"
+            }
+          }
+        ]
+      }
+    },
+    {
+
+      "type": "Microsoft.Network/publicIPAddresses",
+
+      "name": "[variables('publicIPAddressName')]",
+
+      "location": "[resourceGroup().location]",
+
+      "apiVersion": "2017-06-01",
+
+      "properties": {
+
+        "publicIPAllocationMethod": "Dynamic"
+      }
+
+    }
+  ],
+
+  "outputs": {
+    "existingSubnetId": {
+      "type": "string",
+      "value": "[resourceId('Microsoft.Network/virtualNetworks',variables('vnetFESubnetName'))]"      
```

Comment:
This will not return a subnetId - it will return the resourceId of the vnet...  try:

resourceId('Microsoft.Network/virtualNetworks/subnets', variables('vnetName'), variables('vnetFESubnetName'))